src/app/modules/group/http-services/create-group-invitations.service.ts

```diff
@@ -16,15 +22,24 @@ export class CreateGroupInvitationsService {
   createInvitations(
     groupId: string,
     logins : string[]
-  ) : Observable<Map<string, any>>
+  ) : Observable<Map<string, InvitationResult>>
   {
     return this.http
       .post<ActionResponse<Object>>(
         `${environment.apiUrl}/groups/${groupId}/invitations`,
         {logins: logins}, {})
       .pipe(
         map(successData),
-        map(objectToMap)
+        map(function (data: Object): Map<string, InvitationResult> {
+          return new Map<string, InvitationResult>(
+            Object.entries(data).map(
+              ([key, value]) => {
+                const result = InvitationResult[value as keyof typeof InvitationResult];
+                if (result == undefined)
+                  throw new Error(`Invitation of user ${key} returned an unexpected result`);
+                return [key, value];
```

Comment:
Ah I didn't know this weird syntax `InvitationResult[value as keyof typeof InvitationResult];`
Actually, I have not been clear but the service returns (following the API) `enum: [cycle, invalid, success, unchanged, not_found]` so you have to map these input values to the enum (cycle and invalid can be both mapped to error, the UI does not need to know more). So I expect more a switch-case which return the different values of the enum.